### PR TITLE
Ledger API test tool: test against legacy and preview version.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 1]
```diff
@@ -47,97 +48,117 @@ sh_binary(
...
+            "--exclude=ContractKeysIT,ContractKeysIT:CKFetchOrLookup,ContractKeysIT:CKNoFetchUndisclosed,ContractKeysIT:CKMaintainerScoped" +
+            ",ParticipantPruningIT" +  # see "conformance-test-participant-pruning" below
+            ",ConfigManagementServiceIT,LedgerConfigurationServiceIT" +  # dynamic config management not supported by Canton
+            ",ClosedWorldIT" +  # Canton currently fails this test with a different error (missing namespace in "unallocated" party id)
+            # The following tests fail because Canton raises a different error (see https://github.com/digital-asset/daml/pull/8642)
+            ",PartyManagementServiceIT:PMRejectLongPartyHints,PartyManagementServiceIT:PMRejectInvalidPartyHints",
+        ],
+    )
+    for lf_version in lf_versions_aggregate([
+        # TODO: uncomment the following line and drop the "legacy" as soon canton-test-runner support LF 1.11 and newers
```
Comment (remyhaemmerle-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:87]: cc @oliverse-da The 2 TODOs in this file are for you.

**Review by stefanobaghino-da — APPROVED** [review]

File: ledger/ledger-on-memory/BUILD.bazel [hunk 2]
```diff
@@ -128,18 +129,26 @@ da_scala_binary(
...
+        test_tool_args = [
+            "--verbose",
+        ],
+    )
+    for lf_version in lf_versions_aggregate([
+        "stable",
+        "latest",
+        "preview",
+    ])
+]
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply ledger/ledger-on-memory/BUILD.bazel:151]: any idea for naming the rule ?

File: ledger/sandbox/BUILD.bazel [hunk 3]
```diff
@@ -255,7 +256,7 @@ server_conformance_test(
 )
 
 server_conformance_test(
-    name = "next-conformance-test-wall-clock-time",
+    name = "next-conformance-test",
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply ledger/sandbox/BUILD.bazel]: reverted.

File: ledger/ledger-on-memory/BUILD.bazel (hunk 2 above)
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply ledger/ledger-on-memory/BUILD.bazel]: done.

File: ledger/ledger-api-test-tool/conformance.bzl [hunk 4]
```diff
@@ -16,29 +17,31 @@ def conformance_test(
...
+                server_args = server_args,
+                tags = [
+                    "dont-run-on-darwin",
+                    "exclusive",
+                ] + tags,
+                flaky = flaky,
+            )
 
-def server_conformance_test(name, servers, server_args = [], test_tool_args = [], flaky = False):
+def server_conformance_test(name, servers, server_args = [], test_tool_args = [], flaky = False, lf_versions = ["stable", "latest", "preview"]):
```
Comment (stefanobaghino-da) [inline ledger/ledger-api-test-tool/conformance.bzl:42]: Should `lf_versions` here be passed down to the call to the `conformance_test` rule?

**Review by stefanobaghino-da — COMMENTED** [review]
Thanks for addressing my remarks. 🙇🏻 

I think there is one point that requires attention from your latest revision.

**Pull request merged**